circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/CopyJob.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) 2016-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.s3s3copier;
+
+import com.amazonaws.services.s3.model.CopyObjectRequest;
+import com.amazonaws.services.s3.transfer.Copy;
+import com.amazonaws.services.s3.transfer.internal.TransferStateChangeListener;
+
+public class CopyJob {
+  private Copy copy;
+  private CopyObjectRequest copyObjectRequest;
```

Comment:
I think these two fields need to be their own object: `CopyJobRequest`.  Then you can get have just one constructor `CopyJob(copy, copyJobRequest)`. I think that will look nicer in the S3Copier, first create CopyJobRequests then from those create CopyJobs.

Reply:
Thanks, that does clean it up a little. Now there no need for `CopyJob`s with a null field.